pkg/analyze/configmap.go

```diff
@@ -0,0 +1,67 @@
+package analyzer
+
+import (
+	"encoding/json"
+	"fmt"
+
+	troubleshootv1beta2 "github.com/replicatedhq/troubleshoot/pkg/apis/troubleshoot/v1beta2"
+	"github.com/replicatedhq/troubleshoot/pkg/collect"
+)
+
+func analyzeConfigMap(analyzer *troubleshootv1beta2.AnalyzeConfigMap, getCollectedFileContents func(string) ([]byte, error)) (*AnalyzeResult, error) {
+	configMapData, err := getCollectedFileContents(fmt.Sprintf("configMaps/%s/%s.json", analyzer.Namespace, analyzer.ConfigMapName))
+	if err != nil {
+		return nil, err
+	}
+
+	var foundConfigMap collect.ConfigMapOutput
+	if err := json.Unmarshal(configMapData, &foundConfigMap); err != nil {
+		return nil, err
+	}
+
+	title := analyzer.CheckName
+	if title == "" {
+		title = fmt.Sprintf("ConfigMap %s", analyzer.ConfigMapName)
+	}
+
+	result := AnalyzeResult{
+		Title:   title,
+		IconKey: "kubernetes_analyze_configMap",
+		IconURI: "https://troubleshoot.sh/images/analyzer-icons/configMap.svg?w=13&h=16",
+	}
+
+	var failOutcome *troubleshootv1beta2.Outcome
+	for _, outcome := range analyzer.Outcomes {
```

Comment:
could we use the standard analyzer outcomes list code instead, with support for more outcomes than "fail"?

Reply:
or am I just misunderstanding what this analyzer is supposed to be able to analyze

Reply:
good feedback although i didnt write this code. i copied it from secret analyzer. it works as documented now so i think its good enough for this pr.